src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -281,41 +284,45 @@ private async Task PublishFunctionApp(Site functionApp, GitIgnoreParser ignorePa
             var isFunctionAppDedicatedLinux = functionApp.IsLinux && !functionApp.IsDynamic && !functionApp.IsElasticPremium;
 
             // For Python linux apps, we do not support --build remote with --build-native-deps flag
-            if (PublishBuildOption == BuildOption.Remote && BuildNativeDeps)
+            if (PublishBuildOption != BuildOption.Default && BuildNativeDeps)
             {
-                throw new CliException("Cannot use '--build remote' along with '--build-native-deps'");
+                throw new CliException("Cannot use '--build-native-deps' along with '--build' flag");
             }
-            else if (PublishBuildOption == BuildOption.Local ||
-              PublishBuildOption == BuildOption.Container ||
-              PublishBuildOption == BuildOption.None)
+            else if (PublishBuildOption == BuildOption.Container || PublishBuildOption == BuildOption.None)
             {
-                throw new CliException("The --build flag only supports '--build remote'");
+                throw new CliException("The --build flag only supports '--build remote' or '--build local'");
             }
 
             if (GlobalCoreToolsSettings.CurrentWorkerRuntime == WorkerRuntime.python && !functionApp.IsLinux)
             {
                 throw new CliException("Publishing Python functions is only supported for Linux FunctionApps");
             }
 
-            Func<Task<Stream>> zipStreamFactory = () => ZipHelper.GetAppZipFile(functionAppRoot, BuildNativeDeps, PublishBuildOption, NoBuild, ignoreParser, AdditionalPackages, ignoreDotNetCheck: true);
+            // Recommend Linux scm users to use --build remote instead of --build-native-deps
+            if (BuildNativeDeps && functionApp.IsLinux && !string.IsNullOrEmpty(functionApp.ScmUri))
+            {
+                ColoredConsole.WriteLine(Yellow("Recommend using '--build remote' to resolve project dependencies remotely on Azure"));
```

Comment:
@kulkarnisonia16 / @ahmedelnably, should we link them to an aka.ms page, and reword it differently?